File: client/go/src/example/connect/async.go
package connect

import (
	"fmt"
	"context"
	"biostar/service/connect"
)

func (s *ConnectSvc) AddAsyncConnection(connInfos []*connect.AsyncConnectInfo) error {
	req := &connect.AddAsyncConnectionRequest{
		ConnectInfos: connInfos,
	}

	_, err := s.client.AddAsyncConnection(context.Background(), req)

	if err != nil {
		fmt.Printf("Cannot set async connectlion list: %v\n", err)
		return err
	}

	return nil
}


func (s *ConnectSvc) DeleteAsyncConnection(deviceIDs []uint32) error {
	req := &connect.DeleteAsyncConnectionRequest{
		DeviceIDs: deviceIDs,
	}

	_, err := s.client.DeleteAsyncConnection(context.Background(), req)

	if err != nil {
		fmt.Printf("Cannot delete async connection: %v\n", err)
		return err
	}

	return nil
}

File: client/node/example/connect/async.js
const connectMessage = require('../../biostar/service/connect_pb');
const connect = require('./connect');

function addAsyncConnection(connInfos) {
  var req = new connectMessage.AddAsyncConnectionRequest();
  req.setConnectinfosList(connInfos);

  return new Promise((resolve, reject) => {
    connect.getClient().addAsyncConnection(req, (err, response) => {
      if(err) {
        console.error('Cannot add async connection: ', err)
        reject(err);
        return;
      }

      resolve(response);
    });
  });
}

function deleteAsyncConnection(deviceIDs) {
  var req = new connectMessage.DeleteAsyncConnectionRequest();
  req.setDeviceidsList(deviceIDs);

  return new Promise((resolve, reject) => {
    connect.getClient().deleteAsyncConnection(req, (err, response) => {
      if(err) {
        console.error('Cannot delete async connection: ', err)
        reject(err);
        return;
      }

      resolve(response);
    });
  });
}

module.exports.addAsyncConnection = addAsyncConnection;
module.exports.deleteAsyncConnection = deleteAsyncConnection;
